### DISCUSSION TITLE
pyright: import "pandas" could not be resolved

### DISCUSSION THREAD

**Discussion by jychen6688** [zed-industries/zed#6646] [Feature Requests]
When importing a third-party Python library (which I have already downloaded), Zed prompts an error. How should I set it up. For example, in Pandas below, I reported an error while importing, but it was able to run successfully

![image](https://github.com/zed-industries/community/assets/116131272/488a260f-d795-41ef-b85f-4f90335aba1e)

**Comment by krstp** [comment]
Pandas would use the default python distribution, i.e. whatever is in the global definition... so anything you install globally or through `brew` would be accessible.

In order to be more project specific, try to use `pyrightconfig.json` within your project folder, this is what I use atm, i.e. in my case I use virtual env through `pyenv`:
```json
{
  "venvPath": "/Users/<user>/.pyenv/versions",
  "venv": "core"
}
```
As long as your specified virtual env has the package installed it will work. Hope this helps.

**Comment by geoqiao** [comment]
If you are managing Python environment using `Rye`、`Poetry` or `PDM`, add the following to your `pyproject.toml` file:
```toml
[tool.pyright]
venvPath = "."
venv = ".venv"
```
Here is a similar Q&A(https://github.com/zed-industries/community/discussions/1601) that you can refer to.

**Reply by Gufran2809** [reply]
how do i fix it if i'm just using virtualvenv

**Reply by nuriapascual** [reply]
and conda?

**Comment by Set27** [comment]
I am not a python expert and I use asdf for managing python version.

This helps me

pyrightconfig.json
```json
{
  "pythonPath": "/home/<username>/.asdf/installs/python/3.12.4/bin/python",
  "venvPath": "/home/<username>/.asdf/installs/python",
  "venv": "3.12.4"
}

```
